src/components/app/data/hooks/useCouponCodes.js

```diff
@@ -1,34 +1,15 @@
 import { useQuery } from '@tanstack/react-query';
 import useEnterpriseCustomer from './useEnterpriseCustomer';
 import { queryCouponCodes } from '../queries';
-import { hasValidStartExpirationDates } from '../../../../utils/common';
 
 /**
  * Retrieves the coupon codes and transforms their data.
  * @returns {Types.UseQueryResult} The query results for the coupon codes.
  */
 export default function useCouponCodes(queryOptions = {}) {
   const { data: enterpriseCustomer } = useEnterpriseCustomer();
-  const { select, ...queryOptionsRest } = queryOptions;
   return useQuery({
     ...queryCouponCodes(enterpriseCustomer.uuid),
-    ...queryOptionsRest,
-    select: (data) => {
```

Comment:
Moved to service function, so the transform is shared between the UI layer and the route loader layer.